### PR TITLE
Improve and extend common types for data-plane

### PR DESCRIPTION
**Pull request by mikekistler**
This PR updates and extends the common-types file for data-plane with common components that should be useful to many / most data-plane services.

### CODE REVIEW

**Review by JeffreyRichter — APPROVED** [review]

**Comment by johanste** [comment]
Was anyone using these type definitions? If so, the breaking change concern is real for client libraries.

**Comment by mikekistler** [comment]
The only uses of these definitions that I could find are in files in the `specification/maps/data-plane/{Search,Route,Common}/preview/1.0` directories.  I will reach out to the Maps team to confirm that the changes I have made are acceptable.

**Comment by mikekistler** [comment]
@propheel I made some small changes to the common definitions that you added to the specs repo.  Please have a look and let me know if you have any concerns with these changes.

File: specification/common-types/data-plane/v1/types.json [hunk 1]
```diff
@@ -70,5 +77,40 @@
...
+      }
+    }
+  },
+  "parameters": {
+    "ApiVersionParameter": {
+      "name": "api-version",
+      "in": "query",
+      "required": true,
+      "type": "string",
+      "description": "The API version to use for this operation.",
```
Comment (markweitzel) [inline specification/common-types/data-plane/v1/types.json:101]: @mikekistler -- This should be more descriptive, e.g. the same format as our date based version, etc.

**Comment by markweitzel** [comment]
@mikekistler -- What about adding the variables for collections, e.g. skip, top, maxpagesize??

File: specification/common-types/data-plane/v1/types.json [hunk 2]
```diff
@@ -70,5 +77,40 @@
...
+      "minLength": 1
+    }
+  },
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
```
Comment (mikekistler) [inline specification/common-types/data-plane/v1/types.json:108]: Need to investigate what values for authorizationUrl need to be supported.  Could be multiple / many.

File: specification/common-types/data-plane/v1/types.json [hunk 3]
```diff
@@ -41,18 +41,25 @@
           },
           "description": "The error additional info."
         }
```
Comment (mikekistler) [inline specification/common-types/data-plane/v1/types.json:43]: Might want to remove "additionalInfo" -- not really part of the common response schema.

File: specification/common-types/data-plane/v1/types.json (hunk 2 above)
Comment (mikekistler) [reply specification/common-types/data-plane/v1/types.json:108]: This isn’t definitive, but a search of the latest versions of all the API docs in this repo shows only one `authorizationUrl`:
```
>ls -d specification/*/resource-manager/**/stable | while read d; do ls -dT $d/* | tail -1; done | while read d; do grep -h '"authorizationUrl"' $d/*.json; done | sort | uniq -c
 391       "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
```

**Comment by mikekistler** [comment]
Closing.  We should explore other approaches to give proper examples for these types.

**Pull request closed without merging**